<!DOCTYPE html>
<html lang="en-US" dir="ltr">
  <head>
    <meta charset="utf-8">
    <meta name="viewport" content="width=device-width,initial-scale=1">
    <title>Identifying HA mutations in a new sequence | Deep mutational scanning of H5 influenza HA</title>
    <meta name="description" content="Interactive figures and detailed results for deep mutational scanning of the HA from the clade 2.3.4.4b A/American Wigeon/South Carolina/USDA-000345-001/2021 (H5N1) strain.">
    <meta name="generator" content="VitePress v1.2.2">
    <link rel="preload stylesheet" href="/Flu_H5_American-Wigeon_South-Carolina_2021-H5N1_DMS/assets/style.CVvlUCqB.css" as="style">
    
    <script type="module" src="/Flu_H5_American-Wigeon_South-Carolina_2021-H5N1_DMS/assets/app.BVyCr-49.js"></script>
    <link rel="preload" href="/Flu_H5_American-Wigeon_South-Carolina_2021-H5N1_DMS/assets/inter-roman-latin.Di8DUHzh.woff2" as="font" type="font/woff2" crossorigin="">
    <link rel="modulepreload" href="/Flu_H5_American-Wigeon_South-Carolina_2021-H5N1_DMS/assets/chunks/framework.C2BfmhQg.js">
    <link rel="modulepreload" href="/Flu_H5_American-Wigeon_South-Carolina_2021-H5N1_DMS/assets/chunks/theme.CnN700P3.js">
    <link rel="modulepreload" href="/Flu_H5_American-Wigeon_South-Carolina_2021-H5N1_DMS/assets/identifying_ha_mutations.md.BfS8b23v.lean.js">
    <script async src="https://www.googletagmanager.com/gtag/js?id=G-P7HL8Q4F41"></script>
    <script>window.dataLayer=window.dataLayer||[];function gtag(){dataLayer.push(arguments)}gtag("js",new Date),gtag("config","G-P7HL8Q4F41");</script>
    <script id="check-mac-os">document.documentElement.classList.toggle("mac",/Mac|iPhone|iPod|iPad/i.test(navigator.platform));</script>
  </head>
  <body>
    <div id="app"><div class="Layout" data-v-5d98c3a5><!--[--><!--]--><!--[--><span tabindex="-1" data-v-0f60ec36></span><a href="#VPContent" class="VPSkipLink visually-hidden" data-v-0f60ec36> Skip to content </a><!--]--><!----><header class="VPNav" data-v-5d98c3a5 data-v-ae24b3ad><div class="VPNavBar top" data-v-ae24b3ad data-v-ccf7ddec><div class="wrapper" data-v-ccf7ddec><div class="container" data-v-ccf7ddec><div class="title" data-v-ccf7ddec><div class="VPNavBarTitle" data-v-ccf7ddec data-v-ab179fa1><a class="title" href="/Flu_H5_American-Wigeon_South-Carolina_2021-H5N1_DMS/" data-v-ab179fa1><!--[--><!--]--><!----><span data-v-ab179fa1>Deep mutational scanning of H5 influenza HA</span><!--[--><!--]--></a></div></div><div class="content" data-v-ccf7ddec><div class="content-body" data-v-ccf7ddec><!--[--><!--]--><div class="VPNavBarSearch search" data-v-ccf7ddec><!----></div><nav aria-labelledby="main-nav-aria-label" class="VPNavBarMenu menu" data-v-ccf7ddec data-v-7f418b0f><span id="main-nav-aria-label" class="visually-hidden" data-v-7f418b0f>Main Navigation</span><!--[--><!--[--><a class="VPLink link VPNavBarMenuLink" href="/Flu_H5_American-Wigeon_South-Carolina_2021-H5N1_DMS/" tabindex="0" data-v-7f418b0f data-v-9c663999><!--[--><span data-v-9c663999>Home</span><!--]--></a><!--]--><!--[--><a class="VPLink link VPNavBarMenuLink" href="/Flu_H5_American-Wigeon_South-Carolina_2021-H5N1_DMS/appendix.html" target="_self" tabindex="0" data-v-7f418b0f data-v-9c663999><!--[--><span data-v-9c663999>Appendix</span><!--]--></a><!--]--><!--]--></nav><!----><!----><div class="VPSocialLinks VPNavBarSocialLinks social-links" data-v-ccf7ddec data-v-0394ad82 data-v-7bc22406><!--[--><a class="VPSocialLink no-icon" href="https://github.com/dms-vep/Flu_H5_American-Wigeon_South-Carolina_2021-H5N1_DMS" aria-label="github" target="_blank" rel="noopener" data-v-7bc22406 data-v-eee4e7cb><span class="vpi-social-github" /></a><!--]--></div><div class="VPFlyout VPNavBarExtra extra" data-v-ccf7ddec data-v-d0bd9dde data-v-b6c34ac9><button type="button" class="button" aria-haspopup="true" aria-expanded="false" aria-label="extra navigation" data-v-b6c34ac9><span class="vpi-more-horizontal icon" data-v-b6c34ac9></span></button><div class="menu" data-v-b6c34ac9><div class="VPMenu" data-v-b6c34ac9 data-v-e7ea1737><!----><!--[--><!--[--><!----><!----><div class="group" data-v-d0bd9dde><div class="item social-links" data-v-d0bd9dde><div class="VPSocialLinks social-links-list" data-v-d0bd9dde data-v-7bc22406><!--[--><a class="VPSocialLink no-icon" href="https://github.com/dms-vep/Flu_H5_American-Wigeon_South-Carolina_2021-H5N1_DMS" aria-label="github" target="_blank" rel="noopener" data-v-7bc22406 data-v-eee4e7cb><span class="vpi-social-github" /></a><!--]--></div></div></div><!--]--><!--]--></div></div></div><!--[--><!--]--><button type="button" class="VPNavBarHamburger hamburger" aria-label="mobile navigation" aria-expanded="false" aria-controls="VPNavScreen" data-v-ccf7ddec data-v-e5dd9c1c><span class="container" data-v-e5dd9c1c><span class="top" data-v-e5dd9c1c></span><span class="middle" data-v-e5dd9c1c></span><span class="bottom" data-v-e5dd9c1c></span></span></button></div></div></div></div><div class="divider" data-v-ccf7ddec><div class="divider-line" data-v-ccf7ddec></div></div></div><!----></header><div class="VPLocalNav empty fixed" data-v-5d98c3a5 data-v-a6f0e41e><div class="container" data-v-a6f0e41e><!----><div class="VPLocalNavOutlineDropdown" style="--vp-vh:0px;" data-v-a6f0e41e data-v-17a5e62e><button data-v-17a5e62e>Return to top</button><!----></div></div></div><!----><div class="VPContent" id="VPContent" data-v-5d98c3a5 data-v-1428d186><div class="VPDoc has-aside" data-v-1428d186 data-v-39a288b8><!--[--><!--]--><div class="container" data-v-39a288b8><div class="aside" data-v-39a288b8><div class="aside-curtain" data-v-39a288b8></div><div class="aside-container" data-v-39a288b8><div class="aside-content" data-v-39a288b8><div class="VPDocAside" data-v-39a288b8 data-v-3f215769><!--[--><!--]--><!--[--><!--]--><nav aria-labelledby="doc-outline-aria-label" class="VPDocAsideOutline" data-v-3f215769 data-v-a5bbad30><div class="content" data-v-a5bbad30><div class="outline-marker" data-v-a5bbad30></div><div aria-level="2" class="outline-title" id="doc-outline-aria-label" role="heading" data-v-a5bbad30>On this page</div><ul class="VPDocOutlineItem root" data-v-a5bbad30 data-v-b933a997><!--[--><!--]--></ul></div></nav><!--[--><!--]--><div class="spacer" data-v-3f215769></div><!--[--><!--]--><!----><!--[--><!--]--><!--[--><!--]--></div></div></div></div><div class="content" data-v-39a288b8><div class="content-container" data-v-39a288b8><!--[--><!--]--><main class="main" data-v-39a288b8><div style="position:relative;" class="vp-doc _Flu_H5_American-Wigeon_South-Carolina_2021-H5N1_DMS_identifying_ha_mutations" data-v-39a288b8><div><h1 id="identifying-ha-mutations-in-a-new-sequence" tabindex="-1">Identifying HA mutations in a new sequence <a class="header-anchor" href="#identifying-ha-mutations-in-a-new-sequence" aria-label="Permalink to &quot;Identifying HA mutations in a new sequence&quot;">​</a></h1><p>H5Nx influenza is a persistent threat to public health, so it&#39;s important to continuously surveil for mutations in H5 that could enhance its ability to infect humans. Thanks to help from <a href="https://lmoncla.github.io/monclalab/" target="_blank" rel="noreferrer">Louise Moncla</a> and <a href="https://lmoncla.github.io/monclalab/team/JordanOrt/" target="_blank" rel="noreferrer">Jordan Ort</a>, you can easily connect H5 HA mutations identified with <a href="https://clades.nextstrain.org/" target="_blank" rel="noreferrer">Nextclade</a> to their effect on H5 stability, cell entry, receptor usage, and antigenic escape. Here&#39;s how:</p><p><a href="https://clades.nextstrain.org/" target="_blank" rel="noreferrer">Nextclade</a> is a software tool that performs clade assignment, mutation calling, and sequence quality checks on viral sequences. After providing your viral sequences as a <code>.fasta</code> file, Nextclade recommends a relevant &#39;dataset&#39; consisting of a curated phylogeny, reference sequence, gene annotation file, and metadata. The output of Nextclade, including viral mutations, can be downloaded for downstream applications.</p><ol><li><p>Prepare and upload a <code>.fasta</code> file of your viral sequences.</p></li><li><p>Select the appropriate reference dataset. We currently support only the <code>H5Nx clade 2.3.4.4</code> dataset, which includes a sequence (<code>A/American_wigeon/North_Carolina/AH0182517/2022</code>) identical in <strong>amino acid</strong> sequence to the parental H5 HA sequence used to design our DMS library (<code>A/American Wigeon/South Carolina/USDA-000345-001</code>).</p></li><li><p>Identify mutations relative to the parental H5 HA sequence. By default, mutations are identified relative to the reference sequence of the <code>H5Nx clade 2.3.4.4</code> dataset (<code>A/Astrakhan/3212/2020</code>). To change this, select <code>A/American_wigeon/North_Carolina/AH0182517/2022</code> from the <code>Relative to</code> drop down menu:</p><p><img src="/Flu_H5_American-Wigeon_South-Carolina_2021-H5N1_DMS/assets/nextclade.V8itk9RO.png" alt="Dropdown menu"></p></li><li><p>Download the mutations identified by Nextclade. Select the <code>nextclade.csv</code> file under the <code>Export</code> tab. Amino acid mutations relative to the parental strain will be in a column called <code>relativeMutations[&#39;A/American_wigeon/North_Carolina/AH0182517/2022&#39;].aaSubstitutions</code> in the following format:</p><div class="language-text vp-adaptive-theme"><button title="Copy Code" class="copy"></button><span class="lang">text</span><pre class="shiki shiki-themes github-light github-dark vp-code" tabindex="0"><code><span class="line"><span>HA:V11I,HA:T52A,HA:M120L,HA:A156T,HA:A226V,HA:K341R,HA:N491D,HA:V526I</span></span></code></pre></div><p><strong>Note</strong>, the <code>aaSubstitutions</code> column still only has the mutations relative to the default Nextclade reference strain (<code>A/Astrakhan/3212/2020</code>).</p></li><li><p>Convert between HA numbering schemes. The HA mutations from Nextclade are numbered <strong>sequentially</strong>. This is equivalent to the <code>sequential_site</code> column in our <a href="https://github.com/dms-vep/Flu_H5_American-Wigeon_South-Carolina_2021-H5N1_DMS/blob/main/data/site_numbering_map.csv" target="_blank" rel="noreferrer"><code>site_numbering_map.csv</code></a>. The file <a href="https://github.com/dms-vep/Flu_H5_American-Wigeon_South-Carolina_2021-H5N1_DMS/blob/main/data/site_numbering_map.csv" target="_blank" rel="noreferrer"><code>site_numbering_map.csv</code></a> provides a way to convert between Nextclade&#39;s sequential site numbering and other HA formats. For a detailed explanation of HA numbering, refer to <a href="./numbering.html">this page</a>.</p></li><li><p>Search for the effect of mutations in our data. Mutations in our data are numbered using the widely adopted H3 numbering scheme, as shown in the <code>reference_site</code> column in <a href="https://github.com/dms-vep/Flu_H5_American-Wigeon_South-Carolina_2021-H5N1_DMS/blob/main/data/site_numbering_map.csv" target="_blank" rel="noreferrer"><code>site_numbering_map.csv</code></a>. First, convert mutations to the H3 numbering scheme (<code>sequential_site</code> → <code>reference_site</code>), then explore their phenotypes using our <a href="./summary.html">interactive summary plots</a>.</p></li></ol></div></div></main><footer class="VPDocFooter" data-v-39a288b8 data-v-d4a0bba5><!--[--><!--]--><!----><!----></footer><!--[--><!--]--></div></div></div><!--[--><!--]--></div></div><footer class="VPFooter" data-v-5d98c3a5 data-v-e315a0ad><div class="container" data-v-e315a0ad><p class="message" data-v-e315a0ad>Study by <a href="https://www.biorxiv.org/content/10.1101/2024.05.23.595634v1">Dadonaite et al (2014), bioRxiv, DOI 10.1101/2024.05.23.595634</a></p><!----></div></footer><!--[--><!--]--></div></div>
    <script>window.__VP_HASH_MAP__=JSON.parse("{\"stability.md\":\"BvDe8_kj\",\"escape.md\":\"CLfcO_sT\",\"a26_usage.md\":\"Bgh7cPPk\",\"index.md\":\"C3hWL-Q_\",\"readme.md\":\"CaQLRtKM\",\"summary.md\":\"J0K8HyjZ\",\"experiments_and_biosafety.md\":\"pciRDciK\",\"numbering.md\":\"CZx9W_jq\",\"cell_entry.md\":\"B4HunSdE\",\"pipeline_information.md\":\"EU-KACYN\",\"identifying_ha_mutations.md\":\"BfS8b23v\"}");window.__VP_SITE_DATA__=JSON.parse("{\"lang\":\"en-US\",\"dir\":\"ltr\",\"title\":\"Deep mutational scanning of H5 influenza HA\",\"description\":\"Interactive figures and detailed results for deep mutational scanning of the HA from the clade 2.3.4.4b A/American Wigeon/South Carolina/USDA-000345-001/2021 (H5N1) strain.\",\"base\":\"/Flu_H5_American-Wigeon_South-Carolina_2021-H5N1_DMS/\",\"head\":[],\"router\":{\"prefetchLinks\":true},\"appearance\":false,\"themeConfig\":{\"nav\":[{\"text\":\"Home\",\"link\":\"/\"},{\"text\":\"Appendix\",\"link\":\"/appendix\",\"target\":\"_self\"}],\"socialLinks\":[{\"icon\":\"github\",\"link\":\"https://github.com/dms-vep/Flu_H5_American-Wigeon_South-Carolina_2021-H5N1_DMS\"}],\"footer\":{\"message\":\"Study by <a href=\\\"https://www.biorxiv.org/content/10.1101/2024.05.23.595634v1\\\">Dadonaite et al (2014), bioRxiv, DOI 10.1101/2024.05.23.595634</a>\"}},\"locales\":{},\"scrollOffset\":134,\"cleanUrls\":false}");</script>
    
  </body>
</html>