import{_ as i,D as e,c as s,I as t,w as l,a2 as a,o as r}from"./chunks/framework.B0JYoH9M.js";const g=JSON.parse('{"title":"Effects of mutations on pseudovirus entry in 293T cells","description":"","frontmatter":{"aside":false},"headers":[],"relativePath":"cell_entry.md","filePath":"cell_entry.md"}'),c={name:"cell_entry.md"},f=a('<h1 id="effects-of-mutations-on-pseudovirus-entry-in-293t-cells" tabindex="-1">Effects of mutations on pseudovirus entry in 293T cells <a class="header-anchor" href="#effects-of-mutations-on-pseudovirus-entry-in-293t-cells" aria-label="Permalink to &quot;Effects of mutations on pseudovirus entry in 293T cells&quot;">​</a></h1><h2 id="interactive-plot-of-mutation-effects" tabindex="-1">Interactive plot of mutation effects <a class="header-anchor" href="#interactive-plot-of-mutation-effects" aria-label="Permalink to &quot;Interactive plot of mutation effects&quot;">​</a></h2><p>The plot below shows how mutations affect cell entry, and allows you to zoom and mouseover sites and mutations. Click on the expansion box in the upper right of the plot to enlarge it for easier viewing, or <a href="/Flu_H5_American-Wigeon_South-Carolina_2021-H5N1_DMS/htmls/293T_entry_func_effects.html" target="_self">click here</a> to open the plot in a stand-alone window.</p>',3),u=a('<p>Here is an explanation of the key plot elements:</p><ul><li>The zoom bar at the top of the plot shows different regions of HA, and can be used to zoom in on specific sites.</li><li>The line plot summarizes the effects of mutations at each site, using the site smmary statistic specified by the interactive option at the bottom of the plot (eg, sum or mean effect of mutations at a site).</li><li>The heatmap shows how each individual mutation affects cell entry. The <code>x</code>&#39;s indicate the amino-acid identity in the parental HA strain, light gray indicates mutations that were not measured, and dark gray indicates mutations that are filtered out by one of the interactive sliders at the bottom of the plot. If you mouseover points, you can also see the measurement in each individual library, which provides a sense of the reproducibility between experiments.</li><li>The options at the bottom of the plot allow you to adjust what is displayed. For instance, the <em>times_seen</em> slider filters for only mutations seen in at least that many unique variants per-library; larger values will generally give more accurate measurements. Similarly, you can filter by how many different experimental selections the mutation was measured in. You can also select to only show mutations accessible by some number of mutations to the codon in the parental sequence (eg, only accessible by single-nucleotide mutations).</li></ul><h2 id="numerical-values-of-mutation-effects-on-cell-entry" tabindex="-1">Numerical values of mutation effects on cell entry <a class="header-anchor" href="#numerical-values-of-mutation-effects-on-cell-entry" aria-label="Permalink to &quot;Numerical values of mutation effects on cell entry&quot;">​</a></h2><p>The numerical data plotted on this page are available in <a href="https://github.com/dms-vep/Flu_H5_American-Wigeon_South-Carolina_2021-H5N1_DMS/blob/main/results/func_effects/averages/293T_entry_func_effects.csv" target="_blank" rel="noreferrer">this CSV file</a>. Note that these values have <strong>not</strong> been filtered by QC criteria like the <em>times seen</em>, so either make sure you understand the filters in the file or use the pre-filtered numerical values provided in the <a href="./summary.html">summary of phenotype effects</a>.</p>',4);function h(m,d,_,p,y,v){const o=e("Altair"),n=e("Figure");return r(),s("div",null,[f,t(n,{caption:"Interactive plot showing effects of mutations on entry into 293T cells"},{default:l(()=>[t(o,{showShadow:!0,"spec-url":"htmls/293T_entry_func_effects.html"})]),_:1}),u])}const T=i(c,[["render",h]]);export{g as __pageData,T as default};
