import{_ as e,c as t,o as a,a2 as n}from"./chunks/framework.C2BfmhQg.js";const i="/Flu_H5_American-Wigeon_South-Carolina_2021-H5N1_DMS/assets/nextclade.V8itk9RO.png",f=JSON.parse('{"title":"Identifying HA mutations in a new sequence","description":"","frontmatter":{},"headers":[],"relativePath":"identifying_ha_mutations.md","filePath":"identifying_ha_mutations.md"}'),o={name:"identifying_ha_mutations.md"},r=n('<h1 id="identifying-ha-mutations-in-a-new-sequence" tabindex="-1">Identifying HA mutations in a new sequence <a class="header-anchor" href="#identifying-ha-mutations-in-a-new-sequence" aria-label="Permalink to &quot;Identifying HA mutations in a new sequence&quot;">​</a></h1><p>H5Nx influenza is a persistent threat to public health, so it&#39;s important to continuously surveil for mutations in H5 that could enhance its ability to infect humans. Thanks to help from <a href="https://lmoncla.github.io/monclalab/" target="_blank" rel="noreferrer">Louise Moncla</a> and <a href="https://lmoncla.github.io/monclalab/team/JordanOrt/" target="_blank" rel="noreferrer">Jordan Ort</a>, you can easily connect H5 HA mutations identified with <a href="https://clades.nextstrain.org/" target="_blank" rel="noreferrer">Nextclade</a> to their effect on H5 stability, cell entry, receptor usage, and antigenic escape. Here&#39;s how:</p><p><a href="https://clades.nextstrain.org/" target="_blank" rel="noreferrer">Nextclade</a> is a software tool that performs clade assignment, mutation calling, and sequence quality checks on viral sequences. After providing your viral sequences as a <code>.fasta</code> file, Nextclade recommends a relevant &#39;dataset&#39; consisting of a curated phylogeny, reference sequence, gene annotation file, and metadata. The output of Nextclade, including viral mutations, can be downloaded for downstream applications.</p><ol><li><p>Prepare and upload a <code>.fasta</code> file of your viral sequences.</p></li><li><p>Select the appropriate reference dataset. We currently support only the <code>H5Nx clade 2.3.4.4</code> dataset, which includes a sequence (<code>A/American_wigeon/North_Carolina/AH0182517/2022</code>) identical in <strong>amino acid</strong> sequence to the parental H5 HA sequence used to design our DMS library (<code>A/American Wigeon/South Carolina/USDA-000345-001</code>).</p></li><li><p>Identify mutations relative to the parental H5 HA sequence. By default, mutations are identified relative to the reference sequence of the <code>H5Nx clade 2.3.4.4</code> dataset (<code>A/Astrakhan/3212/2020</code>). To change this, select <code>A/American_wigeon/North_Carolina/AH0182517/2022</code> from the <code>Relative to</code> drop down menu:</p><p><img src="'+i+'" alt="Dropdown menu"></p></li><li><p>Download the mutations identified by Nextclade. Select the <code>nextclade.csv</code> file under the <code>Export</code> tab. Amino acid mutations relative to the parental strain will be in a column called <code>relativeMutations[&#39;A/American_wigeon/North_Carolina/AH0182517/2022&#39;].aaSubstitutions</code> in the following format:</p><div class="language-text vp-adaptive-theme"><button title="Copy Code" class="copy"></button><span class="lang">text</span><pre class="shiki shiki-themes github-light github-dark vp-code" tabindex="0"><code><span class="line"><span>HA:V11I,HA:T52A,HA:M120L,HA:A156T,HA:A226V,HA:K341R,HA:N491D,HA:V526I</span></span></code></pre></div><p><strong>Note</strong>, the <code>aaSubstitutions</code> column still only has the mutations relative to the default Nextclade reference strain (<code>A/Astrakhan/3212/2020</code>).</p></li><li><p>Convert between HA numbering schemes. The HA mutations from Nextclade are numbered <strong>sequentially</strong>. This is equivalent to the <code>sequential_site</code> column in our <a href="https://github.com/dms-vep/Flu_H5_American-Wigeon_South-Carolina_2021-H5N1_DMS/blob/main/data/site_numbering_map.csv" target="_blank" rel="noreferrer"><code>site_numbering_map.csv</code></a>. The file <a href="https://github.com/dms-vep/Flu_H5_American-Wigeon_South-Carolina_2021-H5N1_DMS/blob/main/data/site_numbering_map.csv" target="_blank" rel="noreferrer"><code>site_numbering_map.csv</code></a> provides a way to convert between Nextclade&#39;s sequential site numbering and other HA formats. For a detailed explanation of HA numbering, refer to <a href="./numbering.html">this page</a>.</p></li><li><p>Search for the effect of mutations in our data. Mutations in our data are numbered using the widely adopted H3 numbering scheme, as shown in the <code>reference_site</code> column in <a href="https://github.com/dms-vep/Flu_H5_American-Wigeon_South-Carolina_2021-H5N1_DMS/blob/main/data/site_numbering_map.csv" target="_blank" rel="noreferrer"><code>site_numbering_map.csv</code></a>. First, convert mutations to the H3 numbering scheme (<code>sequential_site</code> → <code>reference_site</code>), then explore their phenotypes using our <a href="./summary.html">interactive summary plots</a>.</p></li></ol>',4),s=[r];function c(l,d,u,h,m,p){return a(),t("div",null,s)}const g=e(o,[["render",c]]);export{f as __pageData,g as default};
