import{_ as r,c as l,I as s,w as i,j as e,a as t,a2 as c,D as n,o as u}from"./chunks/framework.8Hwwtx2f.js";const T=JSON.parse('{"title":"Effects of mutations on NeuGc usage","description":"","frontmatter":{"aside":false},"headers":[],"relativePath":"NeuGc_usage.md","filePath":"NeuGc_usage.md"}'),h={name:"NeuGc_usage.md"},f=e("h1",{id:"effects-of-mutations-on-neugc-usage",tabindex:"-1"},[t("Effects of mutations on NeuGc usage "),e("a",{class:"header-anchor",href:"#effects-of-mutations-on-neugc-usage","aria-label":'Permalink to "Effects of mutations on NeuGc usage"'},"​")],-1),m=e("h2",{id:"interactive-plot-of-mutation-effects",tabindex:"-1"},[t("Interactive plot of mutation effects "),e("a",{class:"header-anchor",href:"#interactive-plot-of-mutation-effects","aria-label":'Permalink to "Interactive plot of mutation effects"'},"​")],-1),d=e("p",null,"The plot below shows how mutations affect entry in 293 cells expressing Cytidine Monophospho-N-Acetylneuraminic Acid Hydroxylase (CMAH) versus standard 293 cells. Standard 293 cells lack CMAH enzyme and therefore contain only N-acetylneuraminic acid (NeuAc) on their surface. Cells expressing CMAH modify the NeuAc to form N-glycolylneuraminic acid (NeuGc) during glycan biosynthesis. Some mammals, including humans (from which 293 cells are derived), have lost CMAH enzyme expression while others retain it. Most influenza isolates preferentially bind glycans terminating in NeuAc, however, some strains are known to bind NeuGc. The plots below show effects mutations in H5 influenza HA have on entry in 293 versus 293-CMAH cells and calculate entry score differences to highlight mutations that increase entry in NeuGc glycans containing cells.",-1),p=e("p",null,"You can zoom in on specific HA regions using the zoom bar above the heatmaps. Click on the expansion box in the upper right of the plot to enlarge it for easier viewing.",-1),_=e("p",null,[t("The plot shows the difference in entry in cells expressing exclusively NeuGc-terminating glycans versus those expressing NeuAc-terminating glycans. Positive values indicate mutations that improve NeuGc usage. By default, this plot only shows mutations that increase overall entry into NeuGc terminating sialic acid expressing cells; the reason for this is that difference measurements are sometimes noisy for mutations that are highly deleterious to entry in both NeuGc and NeuAc expressing cells and this plot is designed to specifically show mutations that "),e("strong",null,"increase"),t(" NeuGc usage.")],-1),g=e("p",null,"Note the HA numbering scheme used in this page is for sequential numbering (starting at first Methionine).",-1),y=c('<p>Here is an explanation of the key plot elements:</p><ul><li>The zoom bar at the top of the plot shows different regions of HA, and can be used to zoom in on specific sites.</li><li>The line plot summarizes the effects of mutations at each site, using the site smmary statistic specified by the interactive option at the bottom of the plot (eg, sum or mean effect of mutations at a site).</li><li>The heatmap shows how each individual mutation affects NeuGc usage. The <code>x</code>&#39;s indicate the amino-acid identity in the parental HA strain, light gray indicates mutations that were not measured, and dark gray indicates mutations that are filtered out by one of the interactive sliders at the bottom of the plot. If you mouseover points, you can also see the measurement in each individual library, which provides a sense of the reproducibility between experiments.</li><li>The options at the bottom of the plot allow you to adjust what is displayed. For instance, the <em>times_seen</em> slider filters for only mutations seen in at least that many unique variants per-library; larger values will generally give more accurate measurements. Similarly, you can filter by how many different experimental selections the mutation was measured in or the standard deviation (<em>std</em>) across measurements. You can also show only mutations that have at least some minimal cell entry into 293 or 293-CMAH cells, or at least one of those (<em>best_effect</em>). Those sliders are useful for making sure you are showing mutations that are good for entry in at least one cell line in addition to having a large difference between cell lines. You can also select to only show mutations accessible by some number of mutations to the codon in the parental sequence (eg, only accessible by single-nucleotide mutations), or floor the measurements at zero to show / not show mutations that decrease as well as increase NeuGc usage.</li></ul><h2 id="effects-of-mutations-in-neugc-versus-neuac-cells" tabindex="-1">Effects of mutations in NeuGc versus NeuAc cells <a class="header-anchor" href="#effects-of-mutations-in-neugc-versus-neuac-cells" aria-label="Permalink to &quot;Effects of mutations in NeuGc versus NeuAc cells&quot;">​</a></h2><p>The scatterplot below shows effects of mutations on entry in 293-CMAH (NeuGc) versus 293 (NeuAc) cells. If you hover over each point you will find specific mutation labels. You can decrease change <em>best_effect</em> filter to see measurements for mutations that are quite deleterious for entry in both cells lines, but note those measurements tend to be more noisy.</p>',4),N=e("p",null,"The heatmap below shows how mutations affect entry in 293 (NeuAc) versus 293-CMAH (NeuGc) cells. Positive entry values indicate mutation is beneficial for cell entry and negative entry values indicate mutation is deleterious for entry. Note that mutation effects in these plots are determined by many other factors than just NeuAc vs NeuGc binding (such as protein folding, sialic acid binding, etc.), therefore entry difference (the plot at the top of this page and heatmap at the bottom here) is a better way to isolate NeuGc binding specific effects. Dark grey squares indicate mutations filtered out due to deletarious entry effects and you can adjust those filters using sliders below.",-1),v=e("h2",{id:"numerical-values-of-mutation-effects-neugc-usage",tabindex:"-1"},[t("Numerical values of mutation effects NeuGc usage "),e("a",{class:"header-anchor",href:"#numerical-values-of-mutation-effects-neugc-usage","aria-label":'Permalink to "Numerical values of mutation effects NeuGc usage"'},"​")],-1),b=e("p",null,[t("The numerical data plotted on this page are available in "),e("a",{href:"https://github.com/dms-vep/Flu_H5_American-Wigeon_South-Carolina_2021-H5N1_DMS_NeuGc/blob/master/results/summaries/entry_in_NeuAc_vs_NeuGc_cells.csv",target:"_blank",rel:"noreferrer"},"this CSV file"),t(".")],-1);function w(A,G,H,x,C,S){const a=n("Altair"),o=n("Figure");return u(),l("div",null,[f,m,d,p,_,g,s(o,{caption:"Interactive plot showing effects of mutations that increase NeuGc usage"},{default:i(()=>[s(a,{showShadow:!0,"spec-url":"https://dms-vep.org/Flu_H5_American-Wigeon_South-Carolina_2021-H5N1_DMS_NeuGc/htmls/CMAH_vs_293_entry_diffs.html"})]),_:1}),y,s(o,{caption:"Scatteplot showing effect of each mutation in 293-CMAH (NeuGc) versus 293 (NeuAc) cells"},{default:i(()=>[s(a,{showShadow:!0,"spec-url":"https://dms-vep.org/Flu_H5_American-Wigeon_South-Carolina_2021-H5N1_DMS_NeuGc/htmls/CMAH_vs_293_entry_diffs_corr.html"})]),_:1}),N,s(o,{caption:"Mutation entry effects on 293-CMAH (NeuGc) and 293 (NeuAc) cells"},{default:i(()=>[s(a,{showShadow:!0,"spec-url":"https://dms-vep.org/Flu_H5_American-Wigeon_South-Carolina_2021-H5N1_DMS_NeuGc/htmls/entry_in_NeuAc_vs_NeuGc_cells_overlaid.html"})]),_:1}),v,b])}const k=r(h,[["render",w]]);export{T as __pageData,k as default};
